lib/evaluate/type.h

```diff
@@ -96,7 +96,11 @@ class DynamicType {
   }
   explicit constexpr DynamicType(
       const semantics::DerivedTypeSpec &dt, bool poly = false)
-    : category_{TypeCategory::Derived}, derived_{&dt} {}
+    : category_{TypeCategory::Derived}, derived_{&dt} {
+    if (poly) {
+      kind_ = ClassKind;
+    }
```

Comment:
@klausler, FYI, I had to tweak your change here for polymorphic types to work.

Reply:
Thanks.